### PR TITLE
Update working days calculation

### LINKED ISSUE #2196: Incorrect working days calculation in case of substituted holidays from another year
Some countries (e.g. Ukraine, China) have working days moved to the previous or next year (via _substituted holidays_ scheme). If only the year **from which** non-working day is moved is specified, this change is not reflected.

For example, Ukraine has a non-working day moved from December 28, 1996 (Saturday) to January 2, 1997 (Thursday after New Year). Therefore, December 28, 1996 should be a working day. However, it is not:

```python
>>> import holidays
>>> ua = holidays.country_holidays("UA", years=1996)
>>> ua.is_working_day("1996-12-28")
False
```

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Update working days calculation (if there is substituted holidays from another year) .
Close #2196.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've successfully run `make check`, all checks and tests are green

[contributing-guidelines]: https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/holidays/tree/dev/docs/source

### CODE REVIEW

**Review by PPsyrius — APPROVED** [review]
LGTM

File: holidays/groups/custom.py [hunk 1]
```diff
@@ -18,7 +22,7 @@ class StaticHolidays:
     an external class.
     """
 
-    def __init__(self, cls) -> None:
+    def __init__(self, cls, inter_year: bool = False) -> None:
```
Comment (arkid15r) [inline holidays/groups/custom.py]: I don't think the overhead worth introducing the flag, we can check it for all entities.

File: holidays/groups/custom.py [hunk 2]
```diff
@@ -33,3 +37,13 @@ def __init__(self, cls) -> None:
...
+
+            # Has substituted holidays from another year.
+            if inter_year:
+                self.weekend_workdays = set()
+                for subst_data in getattr(self, "special_public_holidays", {}).values():
+                    for data in _normalize_tuple(subst_data):
+                        if len(data) == 5:
+                            _, _, from_month, from_day, from_year = data
+                            from_date = date(from_year, from_month, from_day)
+                            self.weekend_workdays.add(from_date)
```
Comment (arkid15r) [inline holidays/groups/custom.py]: But this has to be out of the loop.

**Review by arkid15r — COMMENTED** [review]
@KJhellico I've updated the code as per your request.
Please review the changes when you get a chance.

**Review by arkid15r — COMMENTED** [review]
Adding to the queue.
@PPsyrius please review when you get a chance.

**Review by PPsyrius — APPROVED** [review]
LGTM

**Pull request merged**